components/brave_wallet/common/zcash_utils.cc

```diff
@@ -93,4 +170,46 @@ std::vector<uint8_t> ZCashAddressToScriptPubkey(const std::string& address,
   return data;
 }
 
+// https://zips.z.cash/zip-0316#encoding-of-unified-addresses
+std::optional<std::string> ExtractTransparentPart(
+    const std::string& unified_address,
+    bool is_testnet) {
+  auto bech_result = bech32::Decode(unified_address);
+
+  if (bech_result.encoding != bech32::Encoding::BECH32M) {
+    return absl::nullopt;
+  }
+
+  std::string expected_hrp = is_testnet ? "utest" : "u";
+  if (bech_result.hrp != expected_hrp) {
+    return absl::nullopt;
+  }
+
+  std::vector<uint8_t> data;
+  if (!ConvertBits<5, 8, false>([&](unsigned char c) { data.push_back(c); },
+                                bech_result.data.begin(),
+                                bech_result.data.end())) {
+    return std::nullopt;
+  }
+
+  auto reverted = RevertF4Jumble(data);
+  if (!reverted || reverted->size() < 16) {
+    return std::nullopt;
+  }
+
+  auto parts = ParseUnifiedAddress(
+      base::make_span(*reverted).subspan(0, reverted->size() - 16));
+  if (!parts.has_value()) {
+    return std::nullopt;
+  }
```

Comment:
Needs a comment and a constant for what is 16

Reply:
Done